price-feeder/oracle/oracle.go

```diff
@@ -33,15 +56,261 @@ func (o *Oracle) Stop() {
 	<-o.closer.Done()
 }
 
-func (o *Oracle) Start(ctx context.Context) {
+// SetPrices retrieve all the prices from
+// our set of providers as determined in the config,
+// average them out, and update the oracle object
+func (o *Oracle) SetPrices() error {
+
+	providerPrices := make(map[string]map[string]sdk.Dec)
+	wg := new(sync.WaitGroup)
+
+	for providerName, tickerList := range o.providerPairs {
+		var priceProvider provider.Provider
+		wg.Add(1)
+
+		switch providerName {
+		case "binance":
+			priceProvider = provider.NewBinanceProvider()
+		case "kraken":
+			priceProvider = provider.NewKrakenProvider()
+		}
+
+		prices, err := priceProvider.GetTickerPrices(tickerList...)
+		if err != nil {
+			return err
+		}
+
+		providerPrices[providerName] = prices
+		wg.Done()
+	}
+
+	wg.Wait()
+
+	// consolidate the different provider maps into one
+	// for each exchange rate
+	var priceAverages = make(map[string]sdk.Dec)
+	var priceCounts = make(map[string]int)
+
+	for _, provider := range providerPrices {
+		for k, price := range provider {
+			if _, ok := priceAverages[k]; !ok {
+				priceAverages[k] = sdk.NewDec(0)
+			}
+			priceAverages[k] = priceAverages[k].Add(price)
+			priceCounts[k]++
+		}
+	}
+
+	// Average out each one
+	for k, average := range priceAverages {
+		floatToMultiply := float64(1) / float64(priceCounts[k])
```

Comment:
@alexanderbez please tell me there's a better way to go from float64 -> sdk.Dec

Reply:
@adamewozniak See multiline suggestion